### ISSUE TITLE
Error thrown when applying project-config with new site group

### ISSUE THREAD

**Issue by yoannisj** [craftcms/cms#7675]
### Description
Applying project-config changes which include creating a new site in a new site group throws:

```
error: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'groupId' cannot be null
```

It looks like the site is being added before the site goup (see screen-shot below).

![project-config-new-site-error](https://user-images.githubusercontent.com/3032696/110992389-c41f6d80-836d-11eb-9581-67af04670f6b.png)

### Steps to reproduce

1. Create new site group locally
2. Create new site locally and assign it to the new group
3. Push code with updated config/project files to staging/production environment
4. Apply project-config changes on staging/production environment (via CLI or CP)

### Additional info

- Craft version: 3.6.10
- PHP version: 7.4
- Database driver & version: Maria DB 10.4

**Comment by andris-sevcenko** [comment]
@yoannisj is the new site group present in the `config/project/siteGroups` folder?

**Comment by yoannisj** [comment]
@andris-sevcenko , yes I can see it in config/project/siteGroups, but it is absent from the database.

**Comment by andris-sevcenko** [comment]
@yoannisj That's strange. Can you send in your database dump, composer.lock, composer.json files, as well as the `project` folder, you're trying to apply?

If so, please send it to support@craftcms.com and reference this issue!

**Comment by yoannisj** [comment]
Thanks @andris-sevcenko for looking into this, and sorry for the slow response.

When I reported the issue, I observed this multiple times but I worked around it by adding a row for the new site group in the database manually before applying the project config changes.

I just noticed now that my `composer.json` file was using `"minimum-stability": "dev"` and after removing this rather permissive setting, I can't reproduce the issue with the database dumps that I have available today (even for dumps that don't have the manually added site group).

If no-one else observed this, I think it is safe to assume it was related to a non-stable package being installed (maybe even craft cms) and to close the issue at this point.
